packages/editor/src/components/post-publish-panel/index.js

```diff
@@ -114,21 +106,21 @@ export default compose( [
 	withSelect( ( select ) => {
 		const {
 			getCurrentPost,
-			getCurrentPostType,
 			isCurrentPostPublished,
 			isCurrentPostScheduled,
-			isSavingPost,
+			isEditedPostBeingScheduled,
 			isEditedPostDirty,
+			isSavingPost,
 		} = select( 'core/editor' );
 		const { isPublishSidebarEnabled } = select( 'core/editor' );
 		return {
-			postType: getCurrentPostType(),
 			hasPublishAction: get( getCurrentPost(), [ '_links', 'wp:action-publish' ], false ),
-			isPublished: isCurrentPostPublished(),
-			isScheduled: isCurrentPostScheduled(),
-			isSaving: isSavingPost(),
+			isBeingScheduled: isEditedPostBeingScheduled(),
```

Comment:
This was the bug: we were using only `isScheduled` to determine whether a post is scheduled, but we also need to take into account whether the date scheduled is > now (which is what `isBeingScheduled` does).